### DISCUSSION TITLE
FetchContent does not work?

### DISCUSSION THREAD

**Discussion by taehyounpark** [nlohmann/json#4325] [Q&A] [answered]
Sorry if this is actually a general CMake question than the library itself. I'm trying to follow the FetchContent way to pull `nlohmann::json` in to my project:
```cmake
  # nlohmann::json
  include(FetchContent)
  FetchContent_Declare(json URL https://github.com/nlohmann/json/releases/download/v3.11.3/json.tar.xz)
  FetchContent_MakeAvailable(json)
  
  target_link_libraries(
  queryosity_extensions
  INTERFACE queryosity::queryosity
  PRIVATE nlohmann_json::nlohmann_json
)
```
But this does not allow the headers to be found. Only when I also do:
```cmake
target_include_directories(
  queryosity_extensions 
  PUBLIC ${CMAKE_CURRENT_SOURCE_DIR} ${nlohmann_json_SOURCE_DIR}/include
)
```
the compilation works. It was my understanding that the full library name `nlohmann_json::nlohmann_json` should be enough for it to be discovered, am I missing something here? Thanks in advance.

**Comment by nlohmann** [comment, marked as answer]
This is how I use the library in a project:

```cmake
cmake_minimum_required(VERSION 3.20)
project(json_test)

include(FetchContent)

FetchContent_Declare(json URL https://github.com/nlohmann/json/releases/download/v3.11.2/json.tar.xz)
FetchContent_MakeAvailable(json)

add_executable(json_test main.cpp)
target_link_libraries(json_test PUBLIC nlohmann_json::nlohmann_json)
```

and then in `main.cpp`

```cpp
#include <nlohmann/json.hpp>
```

**Reply by taehyounpark** [reply]
So changing `PRIVATE` to `PUBLIC` was the solution (at least for me). According to [this post](https://stackoverflow.com/questions/69783203/examples-of-when-public-private-interface-should-be-used-in-cmake) there seems to be no reason to not do this when we'll pretty much always be working with the JSON objects from this library?
